src/main/java/org/jboss/staxmapper/XMLMapperImpl.java

```diff
@@ -100,10 +108,24 @@ public void deparseDocument(XMLContentWriter contentWriter, XMLStreamWriter stre
         contentWriter.writeContent(new FormattingXMLStreamWriter(streamWriter));
     }
 
-    @SuppressWarnings({ "unchecked" })
+    private XMLElementReader<?> getParser(final QName name) {
+        rootElements.computeIfAbsent(name, qName -> {
+            XMLElementReader<?> reader = rootElementsProvider.get(name).get(); //lazy create of parser
+            return new WeakReference<>(reader);
```

Comment:
provider doesn't necessary have reference to it, provider#get() returns the instance of parser, but doesn't have to hold on to it. i can always create it.
Maybe better method name would be #create and I should probably add some javadoc